source/Octopus.Tentacle.Client/TentacleClient.cs

```diff
@@ -209,6 +211,136 @@ async Task<DataStream> DownloadFileAction(CancellationToken ct)
             }
         }
 
+
+        public async Task<ScriptStatusResponseV3Alpha> StartScript(
+            StartScriptCommandV3Alpha startScriptCommand,
+            ILog logger,
+            CancellationToken scriptExecutionCancellationToken)
+        {
+            //TODO: sast - think about metrics. This isn't right, so we have ignored it for now. But we need to ask what we want to report on in an async event based world.
+            var operationMetricsBuilder = ClientOperationMetricsBuilder.Start();
+
+            var scriptServiceVersion = await DetermineScriptServiceVersionToUse(operationMetricsBuilder, logger, scriptExecutionCancellationToken);
+
+            var factory = new ScriptServiceExecutorFactory(
+                scriptServiceV1,
+                scriptServiceV2,
+                scriptServiceV3Alpha,
+                rpcCallExecutor,
+                operationMetricsBuilder,
+                OnCancellationAbandonCompleteScriptAfter,
+                clientOptions,
+                logger);
+
+            var executor = factory.CreateExecutor(scriptServiceVersion);
+
+            var result = await executor.StartScript(startScriptCommand, scriptExecutionCancellationToken);
+            return result;
+        }
+
+        public async Task<ScriptStatusResponseV3Alpha> GetStatus(
+            GetStatusCommandV3Alpha getStatusCommand,
+            ILog logger,
+            CancellationToken scriptExecutionCancellationToken)
+        {
+            //TODO: sast - think about metrics. This isn't right, so we have ignored it for now. But we need to ask what we want to report on in an async event based world.
+            var operationMetricsBuilder = ClientOperationMetricsBuilder.Start();
+
+            var scriptServiceVersion = await DetermineScriptServiceVersionToUse(operationMetricsBuilder, logger, scriptExecutionCancellationToken);
```

Comment:
Could this be cached within that second object that could be returned by start script. Otherwise we are very much relying on halibut caching to work. If we don't cache this we will need to ensure we don't log too much since `DetermineScriptServiceVersionToUse` might be logging to task log.

Reply:
Yeah, good points 🤔 

The complication is a multi-node environment.

Returning the version in start script would be the most efficient, but it does expose internal concerns to the public 😞 

Like you say, caching does help, but it does mean every server node that does a task (e.g. GetStatus) now has to determine the version again (and log it to the task log 😑 ).

Which then brings us to the idea of reducing what we log when doing this.

@jimmyp , do you have any thoughts on this one?

Reply:
> very much relying on halibut caching to work
I dont understand halibut enough to understand this

> ensure we don't log too much since DetermineScriptServiceVersionToUse might be logging to task log.
I'm wondering if this is something that might need to change about tentacle client. The way I've been thinking about task logging in the new pipeline is that its not something most components should know about. The pipeline will raise events and task logging component somewhere will build up the task log as a read model.

> Returning the version in start script would be the most efficient, but it does expose internal concerns to the public 😞
What exactly is the internal concern being exposed here sorry? In general our multi-node caching mechanism is going to be the `DeploymentExecution` so my first thought is to reach for that if we do need caching. But I'm keen to here more about why we need caching here and what dependancies this would create

Reply:
Is there anything written down anywhere of the goals and concerns tentacle client is supposed to deal with?